core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3523,15 +3523,30 @@ object Stream extends StreamLowPriority {
     * All elements that are available, up to the specified limit,
     * are dequeued and emitted as a single chunk.
     */
-  def fromQueueUnterminated[F[_]: Functor, A](
+  def fromQueueUnterminated[F[_], A](
       queue: QueueSource[F, A],
       limit: Int = Int.MaxValue
-  ): Stream[F, A] =
-    fromQueueNoneTerminatedSingletons_[F, A](
-      queue.take.map(a => Some(a)),
-      queue.tryTake.map(_.map(a => Some(a))),
-      limit
-    )
+  )(implicit F: Functor[F]): Stream[F, A] =
+    F match {
+      case f0: Monad[F] =>
```

Comment:
Yes, you were right: we do need to keep the pattern match. I forgot that `tryTakeN` requires `Monad[F]` because we added that method after-the-fact ...

Reply:
It weren't the problem if we used a conrete `IO` instead of TF, which would arguably make the signatures more comprehensible without any loss of power for 99.9% of use-cases ;) But that is a personal opinion, not relevant here :)